### PR TITLE
improve service handling

### PR DESCRIPTION
**Pull request by hrzlgnm**
- avoid race conditions with psutil.pids()
- get status of topmost montag service process
- kill all children of a service recursively on stop

### CODE REVIEW

File: pydb/services.py [hunk 1]
```diff
@@ -89,23 +80,33 @@ def as_dict_for_monkey_patching_old_psutils(self, attrs=[], ad_value=None):
...
 def get_current_services_status():
     services_status = _get_default_services_status()
-    for pid in pidlist():
-        try:
-            p = psutil.Process(pid)
-            pinfo = p.as_dict(attrs=['name', 'exe', 'username', 'status', 'cmdline'])
-
-            detected_service_name = executionenvironment.is_montag_process(pinfo, names)
+    try:
+        for p in psutil.process_iter(attrs=['name', 'exe', 'username', 'status', 'cmdline']):
```
Comment (dsk7) [inline pydb/services.py]: At least on ubuntu 14.04 (which is quite old, I know) this does not work - the iter command there takes no arguments.
Ubuntu 14.04 uses psutil 1.2.1.

Comment (dsk7) [reply pydb/services.py]: On the other hand, ubuntu 14.04 is reaching EOL just now :)

**Comment by dsk7** [comment]
What issue does this pull request solve?

**Comment by hrzlgnm** [comment]
Fixes issue https://github.com/montaggroup/montag/issues/216

File: pydb/services.py (hunk 1 above)
Comment (hrzlgnm) [reply pydb/services.py]: I'm voting for the `alte Zöpfe abschneiden` here.

Comment (hrzlgnm) [reply pydb/services.py]: we should also remove the monkey patch for as_dict on the way

**Comment by hrzlgnm** [comment]
another approach: we could monkey patch psutil.process_iter to current version with extra arguments if the effor is not that high, the attrs argument has been added in 2017

**Comment by hrzlgnm** [comment]
I proposing following:

1. start fresh from master: fix the race condition by catching `psutil.NoSuchProcess` exception so the services starts on boot with newer ubuntu versions without crashing
2. rework service handling 
- only consider top most service process for status, ignore childs
- kill all children of a service recursively on stop
- provide a facade for psutil so it still works with psutil from ubuntu 14.4 (1.2.1)

**Comment by hrzlgnm** [comment]
superseded by #218

**Pull request closed without merging**